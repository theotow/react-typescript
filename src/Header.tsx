import * as React from 'react';
import styled from 'styled-components';
import Loadable from 'react-loadable';
import Example from './Hooks1';
import * as Context from './context';

const HeaderComp = styled.header`
  text-align: center;
  height: 200px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid red;
`;

const LoadableComponent = Loadable({
  loader: () => import('./Button'),
  loading: () => null
});

function Header() {
  const context = React.useContext(Context.Context);
  return (
    <HeaderComp>
      <Example />
      <LoadableComponent
        onClick={() => context.setSidebar(true)}
        name="button name"
      />
    </HeaderComp>
  );
}

export default Header;
