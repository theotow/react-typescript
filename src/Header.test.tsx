import Header from './Header';
import React from 'react';
import { shallow } from 'enzyme';

test('two plus two is four', () => {
  expect(shallow(<Header />)).toMatchSnapshot();
});
