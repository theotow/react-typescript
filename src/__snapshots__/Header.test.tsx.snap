// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`two plus two is four 1`] = `
<styled.header>
  <h1>
    header
  </h1>
  <Hooks1 />
  <LoadableComponent
    name="button name"
  />
</styled.header>
`;
